vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+---
+title: "Working with `DataCombined` class"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Working with `DataCombined` class}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  warning = FALSE,
+  message = FALSE,
+  dpi = 300,
+  comment = "#>",
+  out.width = "100%",
+  fig.height = 6,
+  fig.width = 8
+)
+```
+
+## Introduction
+
+In Modeling and Simulation (M&S) workflows, we often need to work with multiple 
+observed and/or simulated datasets. Further, some of these datasets can be linked 
+together in a group (e.g., the simulated and the correnspoding observed data)
+for analysis and visualization.
+
+The `DataCombined` class in `{ospsuite}` provides a container to store 
+data, to group them, and to transform them further. The class accepts data 
+coming from `SimulationResults` or as a `DataSet` object (see [Observed data](observed-data.html)).
+
+Let's first generate some simulation results and load observed data.
+
+```{r}
+library(ospsuite)
+
+# Simulation results
+simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+sim <- loadSimulation(simFilePath)
+simResults <- runSimulation(sim)
+outputPath <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
+
+# observed data
+obsData <- lapply(
+  c("ObsDataAciclovir_1.pkml", "ObsDataAciclovir_2.pkml", "ObsDataAciclovir_3.pkml"),
+  function(x) loadDataSetFromPKML(system.file("extdata", x, package = "ospsuite"))
+)
+# Name the elements of the list according to the names of the loaded `DataSet`.
+names(obsData) <- lapply(obsData, function(x) x$name)
+```
+
+Typically, `DataCombined` is used to create figures using the `plotXXX()` functions. 
+The different plot types are described in [Visualizations with `DataCombined`](data-combined-plotting.html). 
+To visualize the different functionalities of `DataCombined` in this document, 
+we will use the `plotIndividualTimeProfile()` function.
+
+## Creating `DataCombined` object
+
+First, we create a new instance of `DataCombined` class.
+
+```{r}
+myDataCombined <- DataCombined$new()
+```
+
+Then we add simulated results to this object using `$addSimulationResults()`:
+
+```{r}
+myDataCombined$addSimulationResults(
+  simulationResults = simResults,
+  quantitiesOrPaths = outputPath,
+  names = "Aciclovir Plasma",
+  groups = "Aciclovir PVB"
+)
+```
+
+Next we add observed data to this object using `$addDataSets()`:
+
+```{r}
+myDataCombined$addDataSets(
+  obsData$`Vergin 1995.Iv`, 
+  groups = "Aciclovir PVB"
+)
```

Comment:
should we use the `names `argument here as well like for the simulation results?

Reply:
No, this is an example where the names must not be provided explicitly.

By default, if no names are provided, then the `name` property of a `DataSet´ or the output path of the simulation result are used as names. For simulation results, the user more likely would like to specify the name.

Reply:
ok